src/node_zlib.cc

```diff
@@ -254,6 +257,19 @@ class ZCtx : public AsyncWrap {
             ctx->err_ = Z_NEED_DICT;
           }
         }
+        if (ctx->strm_.avail_in >= GZIP_MIN_HEADER_SIZE &&
+            ctx->mode_ == GUNZIP) {
+          // Bytes remain in input buffer. Perhaps this is another compressed
+          // member in the same archive, or just trailing garbage.
+          // Check the header to find out.
+          if (*ctx->strm_.next_in != GZIP_HEADER_ID1 ||
+              *(ctx->strm_.next_in + 1) != GZIP_HEADER_ID2) {
```

Comment:
It's a little more idiomatic to write it as `ctx->strm_.next_in[1] != GZIP_HEADER_ID2`.  Suggestion: use indexed notation for the first byte too, for consistency; that's what we do elsewhere.